# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1207082148 14400
#      Tue Apr 01 16:35:48 2008 -0400
# Node ID 3c16e24145647e15720caaae59e02c1d65f54d6e
# Parent  5988e08c1ae6d818c71c6d040e7a8b4b86537d5e
perl: don't provide default value for varargin

diff --git a/scripts/miscellaneous/perl.m b/scripts/miscellaneous/perl.m
--- a/scripts/miscellaneous/perl.m
+++ b/scripts/miscellaneous/perl.m
@@ -21,17 +21,21 @@
 ## @deftypefnx {Function File} {[@var{output}, @var{status}] =} perl (@var{scriptfile}, @var{argument1}, @var{argument2}, ...)
 ## Invoke perl script @var{scriptfile} with possibly a list of
 ## command line arguments.
 ## Returns output in @var{output} and status
 ## in @var{status}.
 ## @seealso{system}
 ## @end deftypefn
 
-function [output, status] = perl (script = "-e ''", varargin = {})
+function [output, status] = perl (script = "-e ''", varargin)
+
+  ## VARARGIN is intialized to {}(1x0) if no additional arguments are
+  ## supplied, so there is no need to check for it, or provide an
+  ## initial value in the argument list of the function definition.
 
   if (ischar (script)
       && ((nargin != 1 && iscellstr (varargin))
 	  || (nargin == 1 && ! isempty (script))))
     [status, output] = system (cstrcat ("perl ", script,
 					sprintf (" %s", varargin{:})));
   else
     error ("perl: invalid arguments");
